Library/Homebrew/cmd/cleanup.rb

```diff
@@ -62,6 +62,10 @@ def cleanup_cache
         cleanup_path(path) { path.unlink }
         next
       end
+      if path.to_s.include?("java_cache")
```

Comment:
You could make the same statement about any cache directory/parent which includes `--` in the name in that case, which seems *considerably* more likely. I'm not sure we should necessarily be devoting an enormous amount of time to protecting people against Homebrew-related Darwinism, of which there are inevitably a bunch of tripwires for in the core.

>  I don't see why we shouldn't use a 100% safe and simple alternative instead of leaving a nonzero (if small) chance of things going completely wrong.

I think it's reasonable to try and strike a balance between protecting everyone under the sun using Homebrew from self-inflicted "Doh" moments and speed.

A rough average across 5 benchmarks of each shows the suggested method is 0.5 to 0.9 seconds slower per run. We've been fussy about less than that elsewhere, many many many many times :sweat: